### DISCUSSION TITLE
Uprising 2: Lead and Destroy does not work at all

### DISCUSSION THREAD

**Discussion by DSN9990** [JHRobotics/softgpu#74] [Q&A]
Hi @JHRobotics, I'm trying to play Uprising 2 through my Win98 SE machine on VMware Workstation 17 but it doesn't seem to work, it crashes on startup with this error:
![Windows 98-2024-04-14-17-14-58](https://github.com/JHRobotics/softgpu/assets/69558812/fba0d4f9-dfc1-4a83-9533-20cd5d3442b1)

Can you help me? I really love this old game. Here's the config and log as well:

[vmware.log](https://github.com/JHRobotics/softgpu/files/14971994/vmware.log)
[Windows 98.vmx.zip](https://github.com/JHRobotics/softgpu/files/14971997/Windows.98.vmx.zip)

**Comment by DSN9990** [comment]
Hi, can you help me please?

**Comment by JHRobotics** [comment]
Install "software renderer" version and insert this key to registry:
```
REGEDIT4

[HKEY_LOCAL_MACHINE\Software\DDSwitcher]
"Uprising 2.exe"="system"
```

This bypass Wine and use driver DirectDraw. The 3DFX version does weird things (like it's using both DirectX and Glide and copy frame buffer between them) - I'll see if it can be fixed in time.

**Reply by DSN9990** [reply]
Yeah, thanks. I'll give it a try. So there is no way to play the 3Dfx version for now?

**Reply by JHRobotics** [reply]
Not now. I'll look into why it doesn't work later as part of compatibility improvements. But now I'm afraid you'll have to stick with SW renderer for while.

**Comment by DSN9990** [comment]
@JHRobotics I have an unrelated question here: is it possible to smooth the edges of fonts on the VMDisp9x driver? I tried setting this on the Display Properties on Win98 SE but this didn't work at all. Maybe this driver does not support this yet or is there an registry setting to enable this?

**Reply by DSN9990** [reply]
Hi? Is there any way to enable font smoothing like I said above?

**Reply by JHRobotics** [reply]
ClearType feature was added to XP[^1], so original 95/98/Me didn't support this. But there is something named Revolutions Pack 9 (http://rp9.win98.nl/) which intergated XP look (and ClearType) to Windows 9x. But I didn't tried this, so I can't more help.

[^1]: And when I migrated 98 -> XP I disable this, because it slowing system down, but was this was still era of CRT screens.

**Reply by DSN9990** [reply]
@JHRobotics No, I'm talking about the "Smooth edges of screen fonts" setting on the Display Properties window.
I tried setting it to on, but it did not work for some reason. So how I can enable this anyway?

**Reply by DSN9990** [reply]
How can I enable this setting like I said above?

**Reply by DSN9990** [reply]
Hi?

**Comment by kjliew** [comment]
> So there is no way to play the 3Dfx version for now?

Use Sven's Glide wrapper for other games.
https://www.svenswrapper.de/english/other.html

Uprising/Uprising2 game engine 3Dfx Glide rendering path uses a rare & uncommon LOD-to-bias blending mode that is quite difficult to implement in the days of FFP GPUs. Outside of dgVoodoo2, this is the ONLY Glide wrapper that specifically designed for these games. It works for qemu-3dfx which I presume would be the similar for SoftGPU.

**Comment by DSN9990** [comment]
Hi @JHRobotics! I'm having some weird issues with Age of Empires on my Win98 guest on VMware Workstation 16.1.2. It crashes with this message when desktop is set to 32bit color:
![Windows 98 SE-2024-04-21-07-23-18](https://github.com/JHRobotics/softgpu/assets/69558812/79a269f7-2156-423f-8fb7-c41abe1d2be0)
Log files:
[Windows 98 SE.zip](https://github.com/JHRobotics/softgpu/files/15051566/Windows.98.SE.zip)

Is there any way to run AOE optimally?

**Reply by DSN9990** [reply]
And what about Uprising 2? Is there any Glide wrapper that works with it?

**Reply by DSN9990** [reply]
Hi? Any help?

**Comment by DSN9990** [comment]
Hi @JHRobotics, I'm having serious performance issues on DX8-9 games on VirtualBox 7.0.16 with vGPU 10. How can I improve performance anyway? If I cannot, can you further optimize the performance of vGPU 10 on VBox 7.0.16?

**Reply by DSN9990** [reply]
Hi? Any response please?
